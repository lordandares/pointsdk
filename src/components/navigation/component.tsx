import React from "react";
import { point, noProfile } from "pointsdk/utils/constants/icons";
import "./style.scss";

export const Navigation: React.FC = () => {
    return (
        <div className="navigation">
            <a href="https://point/">
                <img className="navigation_logo" src={point} alt="point-network" />
            </a>
            <div className="navigation_dropdown">
                <span><h3>Point Network</h3></span>
            </div>

            <a className="navigation_profile_photo_link" href="https://point/">
                <img
                    className="navigation_profile_photo"
                    src={noProfile}
                    alt="point-profile"
                />
            </a>
        </div>
    );
};
