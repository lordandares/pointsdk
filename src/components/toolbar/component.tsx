import React, { useContext } from "react";
import { AppContext } from "@src/context";
// import { Types } from "@src/types/shared";
import {
    ICON_DRIVE,
    ICON_MESSAGING,
    ICON_WALLET,
    ICON_NOTIFICATION,
    ICON_TERMINAL,
    ICON_EMAIL,
} from "@src/utils/constants/icons";
import {
    WALLET,
    DRIVE,
    EMAIL,
    MESSENGER,
    TERMINAL,
    NOTIFICATION,
} from "@src/utils/constants";
import "./style.scss";

export const Toolbar: React.FC = () => {
    const { dispatch } = useContext(AppContext);

    const dispatchToStore = (payload: string) => {
        return dispatch({
            type: "UPDATE_VIEW",
            payload,
        });
    };
    return (
        <div className="toolbar">
            <img
                onClick={() => dispatchToStore(WALLET)}
                className="toolbar_icons"
                src={ICON_WALLET.default}
                alt="point-wallet"
            />
            <img
                onClick={() => dispatchToStore(DRIVE)}
                className="toolbar_icons"
                src={ICON_DRIVE.default}
                alt="point-drive"
            />
            <img
                onClick={() => dispatchToStore(EMAIL)}
                className="toolbar_icons"
                src={ICON_EMAIL.default}
                alt="point-email"
            />
            <img
                onClick={() => dispatchToStore(MESSENGER)}
                className="toolbar_icons"
                src={ICON_MESSAGING.default}
                alt="point-messenger"
            />
            <img
                onClick={() => dispatchToStore(TERMINAL)}
                className="toolbar_icons"
                src={ICON_TERMINAL.default}
                alt="point-terminal"
            />
            <img
                onClick={() => dispatchToStore(NOTIFICATION)}
                className="toolbar_icons"
                src={ICON_NOTIFICATION.default}
                alt="point-notification"
            />
        </div>
    );
};
