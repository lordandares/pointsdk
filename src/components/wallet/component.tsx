import React, { useContext } from "react";
import { buy, sell } from "pointsdk/utils/constants/icons";
import { AppContext } from "pointsdk/context";
import "./style.scss";

export const Wallet: React.FC = () => {
    const { state } = useContext(AppContext);

    return (
        <div className="wallet">
            <span className="wallet_node_address_text">Node Address</span>
            <span className="wallet_node_address">{state.wallet.address}</span>
            <div className="wallet_balance">
                <span className="wallet_balance_point">
                    {state.wallet.funds} POINT
                </span>
                <span className="wallet_balance_usd">
                    ${state.wallet.fundsUsd} USD
                </span>
            </div>
            <div className="wallet_icon_actions">
                <div className="wallet_action">
                    <img
                        className="wallet_icon"
                        src={sell}
                        alt="point-sell"
                    />
                    <span>Sell</span>
                </div>
                <div className="wallet_action">
                    <img
                        className="wallet_icon"
                        src={buy}
                        alt="point-buy"
                    />
                    <span>Buy</span>
                </div>
            </div>
        </div>
    );
};
