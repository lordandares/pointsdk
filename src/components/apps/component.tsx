import React, { useContext } from "react";
import { Wallet } from "pointsdk/components/wallet";
import { AppContext } from "pointsdk/context";
import {
    WALLET,
    DRIVE,
    EMAIL,
    MESSENGER,
    TERMINAL,
    NOTIFICATION,
} from "pointsdk/utils/constants";
import "./style.scss";

export const Apps: React.FC = () => {
    // Sends the `popupMounted` event
    const { state } = useContext(AppContext);
    return (
        <div>
            {state.currentView === WALLET && <Wallet />}
            {state.currentView === DRIVE && <div>Drive</div>}
            {state.currentView === EMAIL && <div>Email</div>}
            {state.currentView === MESSENGER && <div>Messenger</div>}
            {state.currentView === TERMINAL && <div>Terminal</div>}
            {state.currentView === NOTIFICATION && <div>Notification</div>}
        </div>
    );
};
