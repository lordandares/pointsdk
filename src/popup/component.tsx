import React from "react";
import { browser } from "webextension-polyfill-ts";
import { Toolbar } from "@src/components/toolbar";
import { Navigation } from "@src/components/navigation";
import { Apps } from "@src/components/apps";
import { AppProvider } from "@src/context";
import "./style.scss";

export const Popup: React.FC = () => {
    // Sends the `popupMounted` event
    React.useEffect(() => {
        browser.runtime.sendMessage({ popupMounted: true });
    }, []);
    return (
        <AppProvider>
            <div className="popup_container">
                <Navigation />
                <Toolbar />
                <Apps />
            </div>
        </AppProvider>
    );
};
