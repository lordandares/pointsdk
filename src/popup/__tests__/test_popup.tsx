import * as React from "react";
import { Popup } from "../component";
import renderer from "react-test-renderer";

test.skip("component renders", () => {
    const tree = renderer.create(<Popup />).toJSON();
    expect(tree).toMatchSnapshot();
});
