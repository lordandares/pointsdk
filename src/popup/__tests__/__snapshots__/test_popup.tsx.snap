// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<div
  className="popup-container"
>
  <h2>
    Point Network
  </h2>
</div>
`;
